apis/extension/node_reservation.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import (
+	"encoding/json"
+	"math"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	AnnotationNodeReservation = NodeDomainPrefix + "/reservation"
+)
+
+// KoordReserved resource reserved by node.annotation,
+// If node.annotation declares the resources to be reserved, like this:
+//  annotations:
+//    node.koordinator.sh/node-reservation: >-
```

Comment:
should update the comment, node.koordinator.sh/node-reservation -> node.koordinator.sh/reservation